src/core/variable.jl

```diff
@@ -1,210 +1,472 @@
+        ) for (l,i,j) in _PMs.ref(pm, nw, :arcs)
     )
 
-    WI = _PMs.var(pm, nw)[:wi] = JuMP.@variable(pm.model,
-        [b in bus_cnd], base_name="$(nw)_wi",
-        start = _PMs.comp_start_value(b[1] != b[2] ? _PMs.ref(pm, nw, :buspairs, b[1:2]) : _PMs.ref(pm, nw, :bus, b[1]), "wi_start", b[3])
+    if bounded
+        for c in cnds
+            flow_lb, flow_ub = _PMs.ref_calc_branch_flow_bounds(_PMs.ref(pm, nw, :branch), _PMs.ref(pm, nw, :bus), c)
+
+            for arc in _PMs.ref(pm, nw, :arcs)
+                l,i,j = arc
+                if !isinf(flow_lb[l])
+                    JuMP.set_lower_bound(q[arc][c], flow_lb[l])
+                end
+                if !isinf(flow_ub[l])
+                    JuMP.set_upper_bound(q[arc][c], flow_ub[l])
+                end
+            end
+        end
+    end
+
+    for (l,branch) in _PMs.ref(pm, nw, :branch)
+        if haskey(branch, "qf_start")
+            f_idx = (l, branch["f_bus"], branch["t_bus"])
+            JuMP.set_start_value(q[f_idx], branch["qf_start"])
+        end
+        if haskey(branch, "qt_start")
+            t_idx = (l, branch["t_bus"], branch["f_bus"])
+            JuMP.set_start_value(q[t_idx], branch["qt_start"])
+        end
+    end
+
+    report && _PMs.sol_component_value_edge(pm, nw, :branch, :qf, :qt, _PMs.ref(pm, nw, :arcs_from), _PMs.ref(pm, nw, :arcs_to), q)
+end
+
+
+
+# "voltage variables, relaxed form"
+# function variable_mc_voltage(pm::_PMs.AbstractWRModel; kwargs...)
+#     variable_mc_voltage_magnitude_sqr(pm; kwargs...)
+#     variable_mc_voltage_product(pm; kwargs...)
+# end
+
+
+"variable: `w[i] >= 0` for `i` in `buses"
+function variable_mc_voltage_magnitude_sqr(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true)
+    cnds = _PMs.conductor_ids(pm; nw=nw)
+    ncnds = length(cnds)
+
+    w = _PMs.var(pm, nw)[:w] = Dict(i => JuMP.@variable(pm.model,
+            [c in 1:ncnds], base_name="$(nw)_w_$(i)",
+            lower_bound = 0.0,
+            start = comp_start_value(_PMs.ref(pm, nw, :bus, i), "w_start", 1.0)
+        ) for i in _PMs.ids(pm, nw, :bus)
     )
 
     if bounded
-        # Diagonal bounds
-        wr_min, wr_max, wi_min, wi_max = _PMs.ref_calc_voltage_product_bounds(_PMs.ref(pm, nw, :buspairs), cnd)
-        for (i, j) in _PMs.ids(pm, nw, :buspairs)
-            JuMP.set_upper_bound(WR[(i, j, cnd, cnd)], wr_max[(i,j)])
-            JuMP.set_upper_bound(WI[(i, j, cnd, cnd)], wi_max[(i,j)])
-
-            JuMP.set_lower_bound(WR[(i, j, cnd, cnd)], wr_min[(i,j)])
-            JuMP.set_lower_bound(WI[(i, j, cnd, cnd)], wi_min[(i,j)])
+        for (i,bus) in _PMs.ref(pm, nw, :bus), c in cnds
+            JuMP.set_lower_bound(w[i][c], bus["vmin"][c]^2)
+            JuMP.set_upper_bound(w[i][c], bus["vmax"][c]^2)
         end
+    end
+
+    report && _PMs.sol_component_value(pm, nw, :bus, :w, _PMs.ids(pm, nw, :bus), w)
+end
+
+#
+# ""
```

Comment:
I assume this must be for SOCWR? Do you think you will reuse any of this function when you rewrite SOCWR? If not go ahead and remove

Reply:
I removed all SOCWR related stuff that will become obsolete when we add it again (tests, variable creation). I kept the constraints however in comments, as those will probably be largely recycled.